plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php

```diff
@@ -66,6 +66,13 @@ class WC_REST_Products_Controller extends WC_REST_Products_V2_Controller {
 	 */
 	private $exclude_status = array();
 
+	/**
+	 * Stores attachment IDs processed during the current request for potential cleanup.
+	 *
+	 * @var array
+	 */
+	private $processed_attachment_ids_for_request = array();
```

Comment:
While this new property does the job, I wonder if we have other alternative to store the attachment IDs, for example [a request param](https://developer.wordpress.org/reference/classes/wp_rest_request/set_param/). The reasons are:
- Minimizing instance state.
- I think the processed attachment IDs naturally belong to the request data than the REST controller instance. The data is also belong to the request only, so you don't need to populate and cleanup between usage.